android-lib/src/androidTest/java/com/kinvey/androidTest/store/data/DataStoreMultiInsertTest.java

```diff
@@ -0,0 +1,1954 @@
+        assertNotNull(saveCallback.result);
+        assertNull(saveCallback.error);
+        Person resultPerson = (Person) saveCallback.result;
+        assertNotNull(resultPerson);
+        assertEquals(id, resultPerson.getId());
+
+        DefaultKinveyReadCallback findCallbackSync = find(syncStore, LONG_TIMEOUT);
+        assertNotNull(findCallbackSync.result);
+        List<Person> listSync = findCallbackSync.result.getResult();
+        assertEquals(1, listSync.size());
+        Person findPersonSync = listSync.get(0);
+        assertNotNull(findPersonSync);
+        assertEquals(id, findPersonSync.getId());
+
+        DefaultKinveyReadCallback findCallbackNet = find(netStore, LONG_TIMEOUT);
+        assertNotNull(findCallbackNet.result);
+        List<Person> listNet = findCallbackNet.result.getResult();
+        assertEquals(1, listNet.size());
+        Person findPersonNet = listNet.get(0);
+        assertNotNull(findPersonNet);
+        assertEquals(id, findPersonNet.getId());
+    }
+
+    @Test
+    public void testSaveWithConnectivityErrorAuto() throws InterruptedException {
+        print("should send with connectivity error");
+        // create an item that has no _id property with invalid _geoloc params
+        // call save() with it -mock it for connectivity error
+        // call find() with syncstore, should return the item from step 1
+        // call pendingSyncEntities(), should return the item from step 1 with POST operation
+
+        Person person = new Person(TEST_USERNAME);
+        person.setGeoloc(ERR_GEOLOC);
+
+        DataStore<Person> autoStore = DataStore.collection(Person.COLLECTION, Person.class, StoreType.AUTO, client);
+        DataStore<Person> syncStore = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        clearBackend(autoStore);
+        clearBackend(syncStore);
+        client.getSyncManager().clear(Person.COLLECTION);
+
+        mockInvalidConnection();
+        save(autoStore, person);
+        cancelMockInvalidConnection();
+
+        DefaultKinveyReadCallback findCallback = find(syncStore, LONG_TIMEOUT);
+        assertNotNull(findCallback.result);
+        List<Person> list = findCallback.result.getResult();
+        assertEquals(1, list.size());
+        assertNotNull(list.get(0).getId());
+
+        List<SyncItem> pendingList = pendingSyncEntities(Person.COLLECTION);
+        assertNotNull(pendingList);
+        assertEquals(1, pendingList.size());
+        SyncItem item = pendingList.get(0);
+        assertNotNull(item);
+        assertEquals(SyncRequest.HttpVerb.POST, item.getRequestMethod());
+    }
+
+    public void testSaveLocallyIfNetworkErrorAuto() throws InterruptedException {
+        print("should save the item with _id locally if network connectivity issue");
+        //create an item with _id
+        //call save with it - mock the request to return connectivity error, should return connec
```

Comment:
I think this would fail, only the first item should be saved in the backend

Reply:
Fixed.